ios/FluentUI/DrawerVnext/SlideOverPanel.swift

```diff
@@ -0,0 +1,174 @@
+
+    // Width occupied by content and spacer combined
+    internal var slideOutPanelWidth: CGFloat = UIScreen.main.bounds.width
+
+    // action executed with background transperent view is tapped
+    internal var actionOnBackgroundTap: (() -> Void)?
+
+    // content view is visible when slide over panel is expanded
+    internal var content: Content
+
+    // opacity used to dim the transparent layer
+    internal var backgroundLayerOpacity: Double = 0.0
+
+    // slide out direction
+    internal var direction: SlideOverDirection = .left
+
+    // content is visible when set to `true`
+    @Binding internal var isOpen: Bool
+
+    // width content view needs to be occupied
+    @Binding internal var preferredContentOffset: CGFloat?
+
+    // configure the apperance of drawer
+    @ObservedObject public var tokens = DrawerTokens()
+
+    private let contentWidthSizeRatio: CGFloat = 0.9
+
+    public var body: some View {
+        HStack {
+            if direction == .right {
+                InteractiveSpacer(defaultBackgroundColor: $tokens.backgroundClearColor)
+                    .onTapGesture (count: 1, perform: actionOnBackgroundTap ?? {})
+            }
+
+            content
+                .frame(width: contentWidth())
+                .shadow(color: tokens.shadowColor.opacity(isOpen ? tokens.shadowOpacity : 0),
+                        radius: tokens.shadowBlur,
+                        x: tokens.shadowDepthX,
+                        y: tokens.shadowDepthY)
+                .offset(x: resolvedContentOffset())
+
+            if direction == .left {
+                InteractiveSpacer(defaultBackgroundColor: $tokens.backgroundClearColor)
+                    .onTapGesture (count: 1, perform: actionOnBackgroundTap ?? {})
+            }
+        }
+        .background(isOpen ? tokens.backgroundDimmedColor.opacity(backgroundLayerOpacity) : tokens.backgroundClearColor)
+    }
+
+    private func resolvedContentOffset() -> CGFloat {
+        // override offset if required
+        if let preferredContentWidth = preferredContentOffset {
+            // parent view wants to take over primarily to conform user drag gesture
+            return preferredContentWidth
+        }
+
+        let offset = isOpen ? expandedConentOffset() : collapsedContentOffset()
+        if direction == .left {
+            return -offset
+        } else {
+            return offset
+        }
+    }
+
+    private func expandedConentOffset() -> CGFloat {
+        return CGSize.zero.width
+    }
+
+    private func collapsedContentOffset() -> CGFloat {
+        return slideOutPanelWidth * contentWidthSizeRatio
+    }
+
+    private func contentWidth() -> CGFloat {
+        return slideOutPanelWidth * contentWidthSizeRatio
+    }
+}
+
+// MARK: - Composite View
+
+public struct InteractiveSpacer: View {
+    @Binding public var defaultBackgroundColor: Color
+
+    public var body: some View {
+        ZStack {
+            defaultBackgroundColor
+        }
```

Comment:
.contentShape(Rectangle())

Adding this modifier here will enable hit testing even if the color is transparent.
This should fix the "tap out of the drawer dismiss bug".

Reply:
this worked, updated !!